packages/core/content-releases/server/src/utils/index.ts

```diff
@@ -1,44 +1,47 @@
 import type { UID, Data, Core } from '@strapi/types';
-import { errors } from '@strapi/utils';
 
 import type { SettingsService } from '../services/settings';
+import type { ReleaseService } from '../services/release';
+import type { ReleaseActionService } from '../services/release-action';
 
 type Services = {
-  release: any;
+  release: ReleaseService;
   'release-validation': any;
   scheduling: any;
-  'release-action': any;
+  'release-action': ReleaseActionService;
   'event-manager': any;
   settings: SettingsService;
 };
 
+interface Action {
+  contentType: UID.ContentType;
+  documentId?: Data.DocumentID;
+  locale?: string;
+}
+
 export const getService = <TName extends keyof Services>(
   name: TName,
   { strapi }: { strapi: Core.Strapi }
 ): Services[TName] => {
   return strapi.plugin('content-releases').service(name);
 };
 
-export const getPopulatedEntry = async (
-  contentTypeUid: string,
-  entryId: Data.ID,
+export const getDraftEntryValidStatus = async (
+  { contentType, documentId, locale }: Action,
   { strapi }: { strapi: Core.Strapi }
 ) => {
   const populateBuilderService = strapi.plugin('content-manager').service('populate-builder');
   // @ts-expect-error - populateBuilderService should be a function but is returning service
-  const populate = await populateBuilderService(contentTypeUid).populateDeep(Infinity).build();
+  const populate = await populateBuilderService(contentType).populateDeep(Infinity).build();
 
-  const entry = await strapi.db.query(contentTypeUid).findOne({
-    where: { id: entryId },
-    populate,
-  });
+  const entry = await getEntry({ contentType, documentId, locale, populate }, { strapi });
 
-  return entry;
+  return getEntryValidStatus(contentType, entry, { strapi });
 };
 
 export const getEntryValidStatus = async (
   contentTypeUid: string,
-  entry: { id: Data.ID; [key: string]: any },
+  entry: any,
   { strapi }: { strapi: Core.Strapi }
 ) => {
```

Comment:
could we name this isEntryValid instead? (if follows more the naming we use to validate data in strapi/core)
And the previous one `isEntryUnpublisheable`